### DISCUSSION TITLE
Can I introduce other UI component libraries, like Echarts

### DISCUSSION THREAD

**Discussion by callback00** [Kitware/trame#451] [Trame/VTK]
I want to implement a gauge chart in Python web, but Vega does not provide a gauge component. Therefore, I want to import other UI component libraries to reduce my workload. Can Echarts component library be introduced into the Trame framework? If so, could you please provide an example?

**Comment by jourdain** [comment]
Trame does support plotly and matplotlib on top of Vega. If none of those handle what you are looking for, we could integrate Echarts into trame by exposing [that library](https://vue-echarts.dev/). But adding that library into trame will require some work that will need your support. If you are interested in funding such integration for the community, please reach out to [Kitware](https://www.kitware.com/contact).
